src/types.ts

```diff
@@ -511,6 +511,179 @@ export type Mutation<R extends Record<string, Any> = Record<string, Any>> =
   | {delete: MutationSelection}
   | {patch: PatchMutationOperation}
 
+/** @public */
+export type Action =
```

Comment:
Re: https://github.com/sanity-io/client/pull/818#discussion_r1611395876
I suggest we make this a union of all the actions, e.g.:
```ts
export type Action =
  | CreateAction
  | ReplaceDraftAction
  | EditAction
  | DeleteAction
  | DiscardAction
  | PublishAction
  | UnpublishAction
```